packages/lyra/src/prefix-tree/trie.ts

```diff
@@ -112,12 +112,12 @@ export class Trie {
     if (!word) return false;
 
     function removeWord(node: TrieNode, _word: string, docID: string): boolean {
-      const [nodeWord /**_docs*/] = node.getWord();
+      const [nodeWord, docIDs] = node.getWord();
 
       if (node.end || (exact && node.end && nodeWord === word)) {
```

Comment:
not related to this PR (rather to #41), but still worth noting: this condition is completely equivalent to just `node.end`: that's because `x || (x && y)` is the same as just `x`. In the specific case, if `node.end` is true, the condition short circuits before the `||`. If it is false, then the part after the `||` is also false, because it's a chain of `&&` including `node.end`. Therefore, the whole `exact` logic can be removed with no effect.

Whether the resulting condition after removing the part after `||` is correct, I am not 100% sure (I would advise having more extensive tests), but for sure it is equivalent to the current one.